mily/utils.py

```diff
@@ -0,0 +1,80 @@
+"""
+Various Qt Utilities
+"""
+import io
+import traceback
+
+from qtpy.QtWidgets import QWidget, QMessageBox
+
+
+def clear_layout(layout):
+    """
+    Clear a QLayout
+
+    This method acts recursively, clearing any child layouts that may be
+    contained within the layout itself. All discovered widgets are marked for
+    deletion.
+
+    Parameters
+    ----------
+    layout : QLayout
+    """
+    while layout.count():
+        child = layout.takeAt(0)
+        if child.widget():
+            child.widget().deleteLater()
+        elif child.layout():
+            clear_layout(child.layout())
+
+
+def reload_widget_stylesheet(widget, cascade=False):
+    """
+    Reload the stylesheet of a QWidget
+
+    When a ``pyqtProperty`` is updated, a widget does not automatically reapply
+    its stylesheet. This is an issue if a stylesheet is conditionally applied
+    based on the ``pyqtProperty`` value. This method handles reapplying the
+    existing stylesheet to guarantee that interface is up-to-date with any
+    changes to the contained properties
+
+    Parameters
+    ----------
+    widget: QWidget
+
+    cascade: bool, optional
+        Any child of the provided widget will also re-update its stylesheet
+    """
+    widget.style().unpolish(widget)
+    widget.style().polish(widget)
+    widget.update()
+    if cascade:
+        for child in widget.children():
+            if isinstance(child, QWidget):
+                reload_widget_stylesheet(child, cascade=True)
+
+
+def raise_to_operator(exc):
+    """
+    Utility function to show a Python Exception in QMessageBox
+
+    The type and representation of the Exception are shown in a pop-up
+    QMessageBox. The entire traceback is available via a drop-down detailed
+    text box in the QMessageBox
+
+    Parameters
+    ----------
+    exc: Exception
+    """
+    # Assemble QMessageBox with Exception details
+    err_msg = QMessageBox()
+    err_msg.setText(f'{exc.__class__.__name__}: {exc}')
+    err_msg.setWindowTitle(type(exc).__name__)
+    err_msg.setIcon(QMessageBox.Critical)
+    # Format traceback as detailed text
+    handle = io.StringIO()
+    traceback.print_tb(exc.__traceback__, file=handle)
+    handle.seek(0)
+    err_msg.setDetailedText(handle.read())
+    # Execute
+    err_msg.exec_()
```

Comment:
Rethinking this, might want to move the `exec` out of this function. Thoughts?

Reply:
At least make it optional, in my opinion

Reply:
why move out? 
You can also make it an option to be modal or not... 
If it is modal you call `exec` otherwise just show.

Reply:
I was thinking more if you wanted to customize slightly. For instance, you might want to add a button to log this error message in some central issue tracker. Then you'd want to do something like:

```python
def my_custom_raiser(exc):
    err_msg = raise_to_operator(exc)
    err_msg.addWidget(...)
    err_msg.exec_()
```

Reply:
It is a nice idea to give the option to customize the dialog!
I also like Ken's input of making it optional too in case folks just want to give a message and that is it...

Reply:
Alternative usages may be nice to add, too, but could be out of the scope of this PR:

```python
def dumb_function():
    with raise_to_operator():
        do_something()
    # no cleanup code necessary here

@raise_to_operator
def dumb_function():
    do_something()
```